examples/software-device-advanced/active_obj.cpp

```diff
@@ -0,0 +1,309 @@
+            RS2_TIMESTAMP_DOMAIN_HARDWARE_CLOCK, int(f.get_frame_number()), vp.get() });
+    }
+}
+
+// Called when the modern API destroys the camera device
+legacy_active_obj::~legacy_active_obj() {
+    alive = false;
+    if (legacy_dev->is_streaming())
+        legacy_dev->stop();
+    
+    if (heart.joinable())
+        heart.join();
+}
+
+// Polls changes to active streams and options
+void legacy_active_obj::heartbeat() {
+    while (alive) {
+        try {
+            // Poll changes in active streams
+            std::map<int, rs2::video_stream_profile> profiles;
+            for (auto &kvp : sensors) {
+                for (auto profile : kvp.second.get_active_streams())
+                    profiles.emplace(profile.unique_id(), profile.as<rs2::video_stream_profile>());
+            }
+            if (profiles != active_profiles) {
+                // Need to update streams.
+                if (legacy_dev->is_streaming()) legacy_dev->stop();
+
+                // Disable all old streams
+                for (int i = 0; i<int(rs::stream::points); ++i) {
+                    auto stream = rs::stream(i);
+                    if (legacy_dev->is_stream_enabled(stream))
+                        legacy_dev->disable_stream(stream);
+                }
+
+                // Enable all active streams
+                for (auto& kvp : profiles) {
+                    auto vp = kvp.second;
+                    legacy_dev->enable_stream(stream_iconv(vp.stream_type()), vp.width(),
+                        vp.height(), fmt_iconv(vp.format()), vp.fps());
+                    legacy_dev->set_frame_callback(stream_iconv(vp.stream_type()), [this](rs::frame f) { 
+                        on_frame(std::move(f)); 
+                    });
+                }
+                active_profiles.swap(profiles);
+                legacy_dev->start();
+            }
+
+            // Poll changes to options
+            for (auto & kvp : cur_options) {
+                rs2_stream stream;
+                rs2_option opt;
+                bool is_writable;
+                std::tie(stream, opt, is_writable) = option_conv(kvp.first);
+                auto sensor = sensors.at(stream);
+                if (is_writable) {
+                    auto new_val = sensor.get_option(opt);
+                    if (new_val != kvp.second) {
+                        kvp.second = new_val;
+                        if (opt == RS2_OPTION_DEPTH_UNITS) new_val *= 1e6; // Legacy API uses micrometers, Modern API uses meters.
+                        legacy_dev->set_option(kvp.first, new_val);
+
+                        // Setting these options causes the auto control to change.
+                        if (kvp.first == rs::option::color_exposure) {
+                            cur_options[rs::option::color_enable_auto_exposure] = 0;
+                            sensor.set_option(RS2_OPTION_ENABLE_AUTO_EXPOSURE, 0);
```

Comment:
Not clear where the `sensor`'s affinity is set to prevent mix depth/color.
Additionally, can you add sensor.support(... query for safety for all the options listed?

Reply:
it's hardcoded into `option_conv`. each legacy option is mapped to a modern (sensor, option) pair. The same function is used to register all the options, and only for which registration succeeds are saved in the cur_options vector which is being iterated.